Add lambda agentcore invocation tutorial

didhd opened the pull request:
# Amazon Bedrock AgentCore Samples Pull Request

> [!IMPORTANT]  
> 1. We strictly follow a issue-first approach, please first open an [issue](https://github.com/awslabs/amazon-bedrock-agentcore-samples/issues) relating to this Pull Request.
> 2. Once this Pull Request is ready for review please attach `review ready` label to it. Only PRs with `review ready` will be reviewed.

**Issue number**: N/A

### Concise description of the PR

Add Lambda AgentCore invocation tutorial with CloudWatch observability to the observability examples.

This PR adds a complete tutorial demonstrating how to invoke Strands agents hosted on Amazon Bedrock AgentCore Runtime from AWS Lambda functions with full CloudWatch observability enabled.

Changes include:

- Complete Lambda AgentCore invocation tutorial in 05-Lambda-AgentCore-invocation directory
- Lambda function code for invoking AgentCore Runtime agents (lambda_agentcore_invoker.py)
- Dockerfile for Lambda container deployment
- Comprehensive Jupyter notebook with step-by-step tutorial (agentcore_observability_lambda.ipynb)
- README.md with architecture, prerequisites, usage, and cleanup documentation
- Updated parent README.md to include this tutorial in the project structure

### User experience

**Before this change:**
- Users had no example of how to invoke AgentCore Runtime agents from Lambda functions
- No guidance on integrating Lambda with AgentCore Runtime and CloudWatch observability
- Missing example of using multiple MCP servers in a Lambda-invoked agent

**After this change:**
- Users can learn how to build serverless architectures with Lambda and AgentCore Runtime
- Complete working example with Lambda function, MCP agent, and Dockerfile
- Step-by-step tutorial showing X-Ray tracing and CloudWatch GenAI Observability integration
- Clear documentation of architecture, setup, and usage patterns

## Checklist

- [x] I have reviewed the [contributing guidelines](https://github.com/awslabs/amazon-bedrock-agentcore-samples/blob/main/CONTRIBUTING.md)
- [x] Add your name to [CONTRIBUTORS.md](https://github.com/awslabs/amazon-bedrock-agentcore-samples/blob/main/CONTRIBUTORS.md)
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/awslabs/amazon-bedrock-agentcore-samples/pulls) for the same update/change?
- [ ] Are you uploading a dataset?
- [x] Have you documented **Introduction**, **Architecture Diagram**, **Prerequisites**, **Usage**, **Sample Prompts**, and **Clean Up** steps in your example README?
- [x] I agree to resolve any [issues](https://github.com/awslabs/amazon-bedrock-agent-samples/issues) created for this example in the future.
- [x] I have performed a self-review of this change
- [x] Changes have been tested
- [x] Changes are documented

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/amazon-bedrock-agentcore-samples/blob/main/LICENSE).

mvangara10 commented on 01-tutorials/06-AgentCore-observability/05-Lambda-AgentCore-invocation/lambda_agentcore_invoker.py:
01-tutorials/06-AgentCore-observability/05-Lambda-AgentCore-invocation/lambda_agentcore_invoker.py:26:15: F541 f-string without any placeholders

mxiamxia commented on 01-tutorials/06-AgentCore-observability/05-Lambda-AgentCore-invocation/README.md:
```diff
@@ -0,0 +1,144 @@
+4. Follow the step-by-step instructions in the notebook to:
+   - Configure CloudWatch Transaction Search
+   - Create and deploy the MCP agent
+   - Build and deploy the Lambda function
```
To display Lambda Invoker Node in Trace view, another separate step needs to be set up to enable Active Tracing for Lambda function.

https://docs.aws.amazon.com/lambda/latest/dg/services-xray.html#services-xray-api

mxiamxia replied:
NVM. Found it is configured as part of Lambda deployment.

didhd commented:
@mvangara10 @mxiamxia Thanks for the detailed feedback! I've addressed all the comments and pushed an update. Regarding the Application Signals Lambda Layer question - I have X-Ray Active Tracing configured but didn't add the Layer. Let me know if that's needed for full Gen AI Observability integration and I'll add it. Please let me know if I missed anything or if there are any other concerns. Thanks again for the thorough review!

mxiamxia commented:
> I have X-Ray Active Tracing configured but didn't add the Layer. Let me know if that's needed for full Gen AI Observability integration and I'll add it.

Could you confirm what the final trace view looks like when the layer isn’t enabled on the Lambda invoker function? I think without the layer, the trace won’t be able to connect AWS Lambda with AgentCore Runtime as described in the README.

didhd commented:
@mxiamxia @mvangara10 
<img width="1209" height="761" alt="Screenshot 2025-10-20 at 11 19 05 PM" src="https://github.com/user-attachments/assets/1f15fdc0-a0e8-4330-8a82-31a6645fbdbe" />

Updated code and tested enabled with Lambda layer - Thank you so much for the detailed feedback!

Can you review this again? 

Thanks!
